rtsf-at-checkout-event-reconciler/events/process_events.go

```diff
@@ -34,98 +40,80 @@ var NextRFIDData = []RFIDEventEntry{}
 var firstBasketOpenComplete = false
 var afterPaymentSuccess = false
 
-func ProcessCheckoutEvents(edgexcontext *appcontext.Context, params ...interface{}) (bool, interface{}) {
+func (eventsProcessing *EventsProcessor) ProcessCheckoutEvents(edgexcontext interfaces.AppFunctionContext, param interface{}) (bool, interface{}) {
+	lc := edgexcontext.LoggingClient()
 
+	devicePos := eventsProcessing.ProcessConfig.DevicePos
 
-	if len(params) < 1 {
-		edgexcontext.LoggingClient.Error("Didn't receive an event")
-		return false, nil
-	}
+	deviceScale := eventsProcessing.ProcessConfig.DeviceScale
 
-	appsettings := edgexcontext.Configuration.ApplicationSettings
-	devicePos, ok := appsettings["DevicePos"]
-	if !ok {
-		edgexcontext.LoggingClient.Error("DevicePos setting not found")
-		return false, nil
-	}
-	deviceScale, ok := appsettings["DeviceScale"]
-	if !ok {
-		edgexcontext.LoggingClient.Error("DeviceScale setting not found")
-		return false, nil
-	}
-	deviceCV, ok := appsettings["DeviceCV"]
-	if !ok {
-		edgexcontext.LoggingClient.Error("DeviceCV setting not found")
-		return false, nil
-	}
-	deviceRFID, ok := appsettings["DeviceRFID"]
-	if !ok {
-		edgexcontext.LoggingClient.Error("DeviceRFID setting not found")
-		return false, nil
-	}
+	deviceCV := eventsProcessing.ProcessConfig.DeviceCV
+
+	deviceRFID := eventsProcessing.ProcessConfig.DeviceRFID
 
-	result, _ := params[0].(models.Event)
+	result, _ := param.(models.Event)
 	for _, reading := range result.Readings {
-		eventName := reading.Name
-		edgexcontext.LoggingClient.Debug(fmt.Sprintf("Processing Checkout Event: %s", eventName))
+		readingData := reading.(models.ObjectReading)
+		eventName := readingData.ResourceName
+		lc.Debugf("Processing Checkout Event: %s", eventName)
 		eventOk := checkEventOrderValid(eventName, edgexcontext)
 		if !eventOk {
-			edgexcontext.LoggingClient.Error(fmt.Sprintf("Error: event occurred out of order: %v", eventName))
+			lc.Errorf("Error: event occurred out of order: %v", eventName)
 			continue
 		}
 
-		switch reading.Device {
-		case devicePos+"-rest", devicePos+"-mqtt":
-			processDevicePosReading(reading, edgexcontext)
+		switch readingData.DeviceName {
+		case devicePos + "-rest", devicePos + "-mqtt":
+			eventsProcessing.processDevicePosReading(readingData, edgexcontext)
```

Comment:
```suggestion
			eventsProcessing.processDevicePosReading(objectReading.ObjectValue, edgexcontext)
```

Reply:
processDevicePosReading() uses fields from BaseReading, such as DeviceName, so I passed in BaseReading instead